src/charm_state.py

```diff
@@ -71,6 +71,7 @@ class CharmState:  # pylint: disable=too-many-instance-attributes
         port: The port of the outgoing SMTP relay.
         user: The SMTP AUTH user to use for the outgoing SMTP relay.
         password: The SMTP AUTH password to use for the outgoing SMTP relay.
+        password_id: The secret ID where the SMTP AUTH password for the SMTP relay is stored.
```

Comment:
Should not the password_id also be added to the `SmtpIntegratorConfig` for validation?

Reply:
It's not part of the config, but a derived value, so I've kept it out of the `SmtpIntegratorConfig`